addons/mail/static/src/core/common/chat_window_model.js

```diff
@@ -52,11 +52,12 @@ export class ChatWindow extends Record {
         }
         await this._onClose(options);
         this.delete();
-        if (escape && indexAsOpened !== -1 && chatHub.actuallyOpened.length > 0) {
-            if (indexAsOpened === chatHub.actuallyOpened.length - 1) {
-                chatHub.actuallyOpened.at(indexAsOpened - 1).focus();
+        const actuallyOpenedLength = chatHub.actuallyOpened.length;
+        if (escape && indexAsOpened !== -1 && actuallyOpenedLength > 0) {
+            if (indexAsOpened === 0) {
+                chatHub.actuallyOpened.at(indexAsOpened % actuallyOpenedLength).focus();
             } else {
-                chatHub.actuallyOpened.at(indexAsOpened).focus();
+                chatHub.actuallyOpened.at((indexAsOpened - 1) % actuallyOpenedLength).focus();
```

Comment:
Why do we even use modulo?

It looks like `indexAsOpened - 1` will always be smaller than `actuallyOpenedLength`?

Reply:
If it was 0, then we focus 0, if it was anything else, it's just last index - 1?